docs/dev/core-refactor.md

```diff
@@ -0,0 +1,71 @@
+# 1. Overview
+
+Improve extensibility of the SQL plugin, OpenSearch functions need to be moved from `core` module and moved to the `opensearch` module. This will prevent other data sources from being able to parse queries with OpenSearch specific functions like `MATCH_QUERY`.
+
+# 2. Requirements
+
+Resolve https://github.com/opensearch-project/sql/issues/811
```

Comment:
Please, add more info. Probably, describe a user story with this feature implemented: plugin for plugin case. It also add dependency clean up and extensibility.